Assets/MRTK/Core/Inspectors/Utilities/SpeechKeywordUtility.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    public static class SpeechKeywordUtility
+    {
+        private static readonly GUIContent SpeechCommandsLabel = new GUIContent("Speech Command", "Speech keyword to use, pulled from MRTK/Input/Speech Commands Profile");
+
+        /// <summary>
+        /// Renders a dropdown with all keywords in the active <see cref="Input.MixedRealitySpeechCommandsProfile"/>.
+        /// </summary>
+        /// <param name="property">The string property representing the selected keyword.</param>
+        public static void RenderKeywords(SerializedProperty property, Rect rect = default(Rect), GUIContent content = null) => RenderKeywordsExcept(null, property, rect, content);
+
+        /// <summary>
+        /// Renders a dropdown with all keywords in the active <see cref="Input.MixedRealitySpeechCommandsProfile"/> except those passed in.
+        /// </summary>
+        /// <param name="keywordsInUse">The keywords the component is currently using.</param>
+        /// <param name="property">The string property representing the selected keyword.</param>
+        public static void RenderKeywordsExcept(string[] keywordsInUse, SerializedProperty property, Rect rect = default(Rect), GUIContent content = null)
+        {
+            IEnumerable<string> availableKeywords = GetDistinctRegisteredKeywords();
+
+            if (keywordsInUse != null && availableKeywords != null)
+            {
+                availableKeywords = availableKeywords.Except(keywordsInUse.Select(keyword => keyword)).ToArray();
+            }
+
+            bool validSpeechKeywords = availableKeywords != null;
```

Comment:
This should also check if availableKeywords is empty
```suggestion
            bool validSpeechKeywords = availableKeywords != null && availableKeywords.Count() > 0;
```

Reply:
Looks good otherwise!

Reply:
Ooh good catch!